data_structures/src/superblock.rs

```diff
@@ -1548,27 +1641,96 @@ mod tests {
+        assert_eq!(sbs.add_vote(&v1, 1), AddSuperBlockVote::ValidWithSameHash);
         assert_eq!(
             sbs.votes_mempool.most_voted_superblock(),
             Some((sb2_hash, 1))
         );
         let mut v2 = SuperBlockVote::new_unsigned(sb2_hash, 1);
         v2.secp256k1_signature.public_key = p1;
-        assert_eq!(sbs.add_vote(&v2), AddSuperBlockVote::ValidWithSameHash);
+        assert_eq!(sbs.add_vote(&v2, 1), AddSuperBlockVote::ValidWithSameHash);
         assert_eq!(
             sbs.votes_mempool.most_voted_superblock(),
             Some((sb2_hash, 2))
         );
         let mut v3 = SuperBlockVote::new_unsigned(sb2_hash, 1);
         v3.secp256k1_signature.public_key = p2;
-        assert_eq!(sbs.add_vote(&v3), AddSuperBlockVote::NotInSigningCommittee);
+        assert_eq!(
+            sbs.add_vote(&v3, 1),
+            AddSuperBlockVote::NotInSigningCommittee
+        );
         assert_eq!(
             sbs.votes_mempool.most_voted_superblock(),
             Some((sb2_hash, 2))
         );
     }
 
+    #[test]
+    fn superblock_vote_may_be_valid() {
+        // Check that no matter the internal state, if we receive a vote that is close to the current index
+        // we mark it as MayBeValid
+        let mut sbs = SuperBlockState::default();
+
+        let v0 = SuperBlockVote::new_unsigned(Hash::SHA256([1; 32]), 10);
+
+        let block_headers = vec![BlockHeader::default()];
+        let pkhs = vec![create_pkh(1)];
+        let keys = vec![create_bn256(1)];
+        let ars_identities = ARSIdentities::new(pkhs.clone());
+        let alt_keys = create_alt_keys(pkhs, keys);
+
+        let genesis_hash = Hash::default();
+        let _sb1 = sbs.build_superblock(
+            &block_headers,
+            ars_identities,
+            100,
+            2,
+            genesis_hash,
+            &alt_keys,
+            None,
+        );
+
+        assert_eq!(sbs.add_vote(&v0, 9), AddSuperBlockVote::MaybeValid);
+    }
+
+    #[test]
+    fn superblock_vote_invalid_index() {
+        // Check that no matter the internal state, if we receive a vote that is far from our current
+        // checkpoint mark it as non-valid
+        let mut sbs = SuperBlockState::default();
+
+        let v0 = SuperBlockVote::new_unsigned(Hash::SHA256([1; 32]), 5);
+
+        let block_headers = vec![BlockHeader::default()];
+        let pkhs = vec![create_pkh(1)];
+        let keys = vec![create_bn256(1)];
+        let ars_identities = ARSIdentities::new(pkhs.clone());
+        let alt_keys = create_alt_keys(pkhs, keys);
+
+        let genesis_hash = Hash::default();
+        let _sb1 = sbs.build_superblock(
+            &block_headers,
+            ars_identities.clone(),
+            100,
+            2,
+            genesis_hash,
+            &alt_keys,
+            None,
+        );
+
+        let _sb2 = sbs.build_superblock(
+            &block_headers,
+            ars_identities,
+            2,
+            1,
```

Comment:
This is the superblock index, it should be the same as the vote index because that used to be marked as "NotInSigningCommittee"

```suggestion
            5,
```

Reply:
Also, you could add this assert:

```rust
         assert_eq!(sbs.add_vote(&v0, 5), AddSuperBlockVote::NotInSigningCommittee); 
```